assignments/prototype-refactor.js

```diff
@@ -1,9 +1,223 @@
+// //       this.stomach.push(someFood);
+// //     }
+// //   };
+
+// //   Person.prototype.poop = function() {
+// //     this.stomach = [];
+// //   };
+
+// //   Person.prototype.toString = function() {
+// //     return `${this.name}, ${this.age}`;
+// //   };
+
+// //----------Class Construction---------
+
+// // class Person {
+// //   constructor(name, age) {
+// //     this.name = name;
+// //     this.age = age;
+// //     this.stomach = [];
+// //   }
+// //   eat(someFood) {
+// //     if (this.stomach.length < 10) {
+// //       this.stomach.push(someFood);
+// //     }
+// //   }
+
+// //   poop() {
+// //     this.stomach = [];
+// //   }
+
+// //   toString() {
+// //     return `${this.name}, ${this.age}`;
+// //   }
+// // }
+
+// // const Fred = new Person("Fred", 37);
+// // Fred.eat();
+// // console.log(Person);
+// // console.log(Fred);
+// // /*
+//     //                     TASK 2
+//     //                     - Write a Car constructor that initializes `model` and `milesPerGallon` from arguments.
+//     //                     - All instances built with Car:
+//     //                     + should initialize with an `tank` at 0
+//     //                     + should initialize with an `odometer` at 0
+//     //                     - Give cars the ability to get fueled with a `.fill(gallons)` method. Add the gallons to `tank`.
+//     // - Give cars ability to `.drive(distance)`. The distance driven:
+//     // + Should cause the `odometer` to go up.
+//     // + Should cause the the `tank` to go down taking `milesPerGallon` into account.
+//     // - A car which runs out of `fuel` while driving can't drive any more distance:
+//     // + The `drive` method should return a string "I ran out of fuel at x miles!" x being `odometer`.
+//     // */
+
+// //    function Car(model, milesPerGallon) {
+// //        this.model = model;
+// //        this.milesPerGallon = milesPerGallon;
+// //        this.tank = 0;
+// //        this.odometer = 0;
+// //     }
+
+// //     Car.prototype.fill = function(gallons) {
+// //         this.tank += gallons;
+// //     };
+
+// //     Car.prototype.drive = function(distance) {
+// //         let needFuel = distance / this.milesPerGallon;
+// //         let milesLeft = this.tank * this.milesPerGallon;
+// //         if (this.tank <= needFuel) {
+// //             this.tank = 0;
+// //             this.odometer = this.odometer + milesLeft;
+// //             return `I ran out of fuel at ${this.odometer} miles!`;
+// //         } else {
+// //             this.odometer += distance;
+// //             this.tank = this.tank - needFuel;
+// //         }
+// //     };
+
+// //----------Class Construction---------
+
+// class Car {
+//   constructor(model, milesPerGallon) {
+//     this.model = model;
+//     this.milesPerGallon = milesPerGallon;
+//     this.tank = 0;
+//     this.odometer = 0;
+//   }
+
+//   fill() {
+//     this.tank += gallons;
+//   }
+
+//   drive() {
```

Comment:
It looks like you're using a variable called `distance` in the body function but you're not passing it in the arguments.